src/db/nedb.js

```diff
@@ -31,13 +32,17 @@ async function connectDB() {
 }
 
 class DBHelper {
-  static async insertOrUpdateTopic(db, topic) {
+  static async insertOrUpdateTopic(db, topic, queryTxid) {
     try {
+      let txid = topic.txid;
+      if (!_.isEmpty(queryTxid)) {
+        txid = queryTxid;
+      }
+
       await db.update(
-        { _id: topic._id },
+        { txid },
         {
           $set: {
-            _id: topic._id,
             txid: topic.txid,
```

Comment:
`txid`

Reply:
fixed

Reply:
just tested this out. we need to set the `txid: topic.txid` because:
1. when I first create a topic/oracle is uses the `approve` txid
2. when the `approve` succeeds, I need to query the topic/oracle by the successful approve txid
3. but I want to update the txid of the topic/oracle to the new pending `createEvent txid` because when the sync updates it, it needs to match the `createEvent txid` and not the `approve txid` or it wont be able to find it

Reply:
@dwalintukan in that case, we should leave `txid` as `null` at first. The rule is to use it as an unique key